src/pages/signup-affiliates/components/_account-personal-details.tsx

```diff
@@ -0,0 +1,333 @@
+            required: true,
+            value_set: setCompanyName,
+            error_set: setCompanyNameErrorMsg,
+        },
+        {
+            id: 'company_registration_number',
+            name: 'company_registration_number',
+            type: 'number',
+            label: localize('_t_Company registration number_t_'),
+            placeholder: 'Company registration number',
+            extra_info: ' ',
+            error: company_registration_error_msg,
+            value: company_registration_number,
+            required: true,
+            value_set: setCompanyRegistrationNumber,
+            error_set: setCompanyRegistrationErrorMsg,
+        },
+        {
+            id: 'website_url',
+            name: 'website_url',
+            type: 'text',
+            label: localize('_t_Website URL_t_'),
+            placeholder: 'Website URL',
+            extra_info: 'Optional',
+            error: website_url_error_msg,
+            value: website_url,
+            required: false,
+            value_set: setWebsiteUrl,
+            error_set: setWebsiteUrlErrorMsg,
+        },
+        {
+            id: 'social_media_url',
+            name: 'social_media_url',
+            type: 'text',
+            label: localize('_t_Social media URL_t_'),
+            placeholder: 'Social media URL',
+            extra_info: 'Optional',
+            error: social_media_url_error_msg,
+            value: social_media_url,
+            required: false,
+            value_set: setSocialMedia,
+            error_set: setSocialMediaErrorMsg,
+        },
+        {
+            id: 'dm-password',
+            name: 'password',
+            type: 'password',
+            label: localize('_t_Password_t_'),
+            placeholder: 'Password',
+            extra_info: ' ',
+            error: password_error_msg,
+            value: password,
+            required: true,
+            value_set: setPassword,
+            error_set: setPasswordErrorMsg,
+        },
+    ]
+    // prevent these characters from the input type='number' tag
+    const input = document.getElementById('company_registration_number')
+    input?.addEventListener('keydown', function (e) {
+        if (['.', '-', '+', 'e'].includes(e.key)) {
+            e.preventDefault()
+        }
+    })
+
+    const validate = is_individual
+        ? username &&
+          first_name &&
+          last_name &&
+          date_birth &&
+          password &&
+          !first_name_error_msg &&
+          !last_name_error_msg &&
+          !password_error_msg
+        : first_name &&
+          last_name &&
+          date_birth &&
+          password &&
+          company_name &&
+          company_registration_number &&
+          !first_name_error_msg &&
+          !last_name_error_msg &&
+          !password_error_msg &&
+          !company_name_error_msg &&
+          !company_registration_error_msg
+
+    useEffect(() => {
+        updateData({
+            ...affiliate_data,
```

Comment:
Why do we use a spread operator here?  🤔 

Also we shouldn’t add the `affiliate_data` and `updateData` as a dependency to this hook?

Reply:
no, we shouldn't, because because we touching only 1 object per step. so by this way we keep prev state and change current object
I'm not prefer this way and better to refactor on `react-hook-form` but no chance before release